### ISSUE TITLE
Feature Request Implement health check probe

### ISSUE THREAD

**Issue by ryanfaircloth** [gaul/s3proxy#427]
Implement a health check probe "/healthz" that returns 200 on get  using a http endpoint not serving s3 i.e 8080 for a get this should return 200 this will prevent infra mon solutions including k8s from killing containers when misconfigured our back end systems fail

**Comment by gaul** [comment]
Does minio or some other private object store have a mechanism that S3Proxy could mimic?

**Comment by ryanfaircloth** [comment]
Yes, its really as simple as an additional "status" endpoint that returns 200 and a json payload { 'status': "ok" } and optionally an error if not ok with a 500.  An example using jetty
https://gist.github.com/jkutner/7b3d154729de6563f7387c26c656233c

The status endpoint will allow k8s to avoid sending requests until the instance is fully started and prevent it from killing the instance if its slow to respond due to backend issues (azure) or just busy

**Comment by gaul** [comment]
I just want to align S3Proxy with a similar implementation instead of creating some proprietary functionality.

**Comment by EraYaN** [comment]
This is essentially how all ASP.NET stuff does it for example, although the content of the actual health check is moot for kubernetes. The only "standard" that is really out there is the Grpc Healthchecks.

Google seems to be the source of the healthz endpoint. So stuff like prometheus also implements it as does ASP.NET Core in most examples. MinIO does something a bit more complex because they want to expose cluster information: https://min.io/docs/minio/linux/operations/monitoring/healthcheck-probe.html But there is nothing stopping you from mapping both `/healthz` and `/minio/health/live` to the same response.

**Comment by missingcharacter** [comment]
@gaul minio does offer a healthcheck https://min.io/docs/minio/linux/operations/monitoring/healthcheck-probe.html and metrics https://min.io/docs/minio/linux/operations/monitoring/metrics-and-alerts.html

**Comment by ppodevlabs** [comment]
any chance this get implemented? it would be really helpfull running on k8s

**Comment by gaul** [comment]
Fixed by #763.
